content/courses/Ada_For_The_C_Embedded_Developer/chapters/07_Reusability.rst

```diff
@@ -491,9 +495,723 @@ Records with discriminants
+actual implementation, we cannot declare objects for it.
+
+We can derive tagged types from an interface and implement the actual
+operations of that interface:
+
+.. code-block:: ada
+
+    type My_Derived is new My_Interface with null record;
+
+    procedure Op (Obj : My_Derived);
+
+Note that we're not using the :ada:`tagged` keyword in the declaration because
+any type derived from an interface is automatically tagged.
+
+Let's look at an example with an interface and two derived tagged types:
+
+.. code:: ada run_button project=Courses.Ada_For_C_Embedded_Dev.Reusability.Interfaces_1
+
+    package P is
+
+       type Display_Interface is interface;
+       procedure Display (D : Display_Interface) is abstract;
+
+       type Small_Display_Type is new Display_Interface with null record;
+       procedure Display (D : Small_Display_Type);
+
+       type Big_Display_Type is new Display_Interface with null record;
+       procedure Display (D : Big_Display_Type);
+
+    end P;
+
+    with Ada.Text_IO; use Ada.Text_IO;
+
+    package body P is
+
+       procedure Display (D : Small_Display_Type) is
+          pragma Unreferenced (D);
+       begin
+          Put_Line ("Using Small_Display_Type");
+       end Display;
+
+       procedure Display (D : Big_Display_Type) is
+          pragma Unreferenced (D);
+       begin
+          Put_Line ("Using Big_Display_Type");
+       end Display;
+
+    end P;
+
+    with P; use P;
+
+    procedure Main is
+       D_Small : Small_Display_Type;
+       D_Big   : Big_Display_Type;
+
+       procedure Dispatching_Display (D : Display_Interface'Class) is
+       begin
+          D.Display;
+       end;
+
+    begin
+       Dispatching_Display (D_Small);
+       Dispatching_Display (D_Big);
+    end Main;
+
+In this example, we have an interface type :ada:`Display_Interface` and two
+tagged types that are derived from :ada:`Display_Interface`:
+:ada:`Small_Display_Type` and :ada:`Big_Display_Type`.
+
+Both types (:ada:`Small_Display_Type` and :ada:`Big_Display_Type`) implement
+the interface by overriding the :ada:`Display` procedure. Then, in the inner
+procedure :ada:`Dispatching_Display` of the :ada:`Main` procedure, we perform
+a dispatching call depending on the actual type of :ada:`D`.
+
+Deriving from multiple interfaces
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+:code-config:`accumulate_code=True`
+
+We may derive a type from multiple interfaces by simply writing
+:ada:`type Derived_T is new T1 and T2 with null record`. For example:
+
+.. code:: ada run_button project=Courses.Ada_For_C_Embedded_Dev.Reusability.Multiple_Interfaces
+
+    package Send_Receive_Pkg is
+
+       type Send_Interface is interface;
+
+       procedure Send (Obj : in out Send_Interface) is abstract;
+
+       type Receive_Interface is interface;
+
+       procedure Receive (Obj : in out Receive_Interface) is abstract;
+
+       type Device_Type is new Send_Interface and Receive_Interface with null record;
```

Comment:
How about changing "Device_Type" to "Transceiver" since it can both send and receive?